programs/quarter_board/main.c

```diff
@@ -0,0 +1,199 @@
+                    else led(RED, 0);
+
+                    break;
+
+        default:    break;
+    }
+
+}
+
+uint8_t *send_data_command(uint16_t command_id, uint8_t data_lenght, uint8_t *buff) {
+    int32_t values = 0;
+    uint8_t i;
+    switch(command_id) {
+        case 0xFF:  values = send_values[0] | (send_values[1] << 16);
+                    led(GREEN, 1);
+                    led(RED, 0);
+                    for(i = 0; i < data_lenght; i++) {
+                        buff[i] = ((values >> (4 * i)) & 0xf);
+                    }
+                    break;
+
+        default:    break;           
+    }
+    return buff;
+}
+
+THD_WORKING_AREA(waCommanderThread, 128);
+THD_FUNCTION(CommanderThread, arg) {
+    (void)arg;
+
+    thread_t *serialp;
+    MESSAGE *messagep;
+    uint16_t cmd, command_id;
+    uint8_t data_buff[MAX_DATA_CHAR_LENGHT];
+    uint8_t data_lenght, i, rw_bit;;
+    while(1) {
+        serialp = chMsgWait();
+        messagep =(MESSAGE *) chMsgGet(serialp);
+        
+        cmd = messagep -> operation;
+
+        command_id = (cmd & 0b11111111111);
+        rw_bit = (((cmd >> OPERATION_BIT_LENGHT) & 1) + 1);
+        data_lenght = (cmd >> (OPERATION_BIT_LENGHT + RW_BIT));
+        
+        if(rw_bit == PROCESS_DATA_FROM_MASTER) {
+            for(i = 0; i < data_lenght; i++) {
+                data_buff[i] = messagep -> data[i];
+            }
+            execute_master_command(command_id, data_buff);
+        } else {
+            send_data_command(command_id, data_lenght, data_buff);
+            for(i = 0; i < data_lenght; i++) {
+                messagep -> data[i] = data_buff[i];
+            }
+        }
+
+        chMsgRelease(serialp, (msg_t)messagep);
+    }
+    chRegSetThreadName("commander");
+    chThdSleepMilliseconds(100);
+}
+
+static const ICUConfig icucfg_1 = {
+    ICU_INPUT_ACTIVE_HIGH,
+    2000000, // 2 MHz timer
+    read_pulse, // pulse received
+    NULL,
+    NULL,
+    ICU_CHANNEL_1,
+    0
+};
+
+static const ICUConfig icucfg_2 = {
+   ICU_INPUT_ACTIVE_HIGH,
+   2000000, // 2 MHz timer
+   read_pulse, // pulse received
+   NULL,
+   NULL,
+   ICU_CHANNEL_2,
+   0
+};
+
+int main(void) {    
+    halInit();
+    chSysInit();
+
+
+    communication_init();
+
+    /*
+     * ICUDriver 1 setup
+     */
+    icuInit();
+    icuStart(&ICUD1, &icucfg_2);
+    icuStartCapture(&ICUD1);
+    icuEnableNotifications(&ICUD1); 
+    palSetPadMode(GPIOA,  9, PAL_MODE_ALTERNATE(2));
+
+    /*
+     * ICUDriver 3 setup
+     */
+    icuStart(&ICUD3, &icucfg_1);
+    icuStartCapture(&ICUD3);
+    icuEnableNotifications(&ICUD3);
+    palSetPadMode(GPIOA, 6, PAL_MODE_ALTERNATE(1));
+    
+    /*
+     * Threads setup
+     */
+    ir_measurementp = chThdCreateStatic(waThread1, sizeof(waThread1), \
+                                        NORMALPRIO, Thread1, NULL);
+    commanderp = chThdCreateStatic(waCommanderThread, sizeof(waCommanderThread), \

```

Comment:
You don't need to escape newline with `\` in C. Please remove them in both this line and the one above.